### DISCUSSION TITLE
WHY Can We NOT Get Rid Of The PRIVATE Shares Option?!

### DISCUSSION THREAD

**Discussion by DeathStalker77** [nicotine-plus/nicotine-plus#3058] [Q&A] [answered]
There are SO many ... jerks .. out there that have their entire collection PRIVATE and will only allow you to get something when you give them something that's basically impossible to get.  WHY even allow it?  The program is for SHARING!  If a user doesn't want an INDIVIDUAL to DL from them, then they can BAN them - VERY simple.  It should not matter whether they are using the orig SLSK, DON'T allow it in N+!  PLEASE!

**Comment by mathiascode** [comment, marked as answer]
I'm done discussing this for the hundredth time. Nicotine+ 3.3.5 will hide private results by default on new installations, similar to the latest SoulseekQt build (i.e. you will have to opt in to see them).

Let's be real. If someone is abusing the feature in order to trade files, they were never going to share anything in the first place. There's a concern that outright removing the feature would just make them find other ways of achieving what they want (custom plugins or forks), while making their results indistinguishable from legit ones. Right now, you can at least hide private results entirely.

**Comment by DeathStalker77** [comment]
Not sure what you mean by "hide private results entirely"?  What it allows is people to abuse the system - they will TAKE what they want while sharing nothing - that's what I'm saying should be PREVENTED.  If they choose to find some other way around it, then at least N+ has done what it can.  That's why I would also like to see what client a user is using (I haven't looked to see if that has been requested as an Enhancement).

I feel we, as responsible users & developers should do everything we can to prevent people from being able to abuse the system.  It's about sharing, not selfishness :-)

**Reply by mathiascode** [reply]
> Not sure what you mean by "hide private results entirely"?

Preferences -> Searches -> Show privately shared files in search results

That option will be disabled by default starting with Nicotine+ 3.3.5.

**Comment by DeathStalker77** [comment]
@mathiascode - how about if someone lists something as PRIVATE, it **ISN'T SHARED** by default?  Put the BURDEN on the "selfish" individuals, instead of the courteous users. :-)  Maybe that won't apply to the orig SLSK client, but it CAN apply to N+.  There are just TOO MANY "jerks" out there, and good users shouldn't have to deal with them when possible.  We don't have an option where we can NAME "Bad Users", so limiting their ability would be the next best thing, imho.  Also, if someone has their ENTIRE share (or even over 50% of their share!) as PRIVATE, they should **NOT** be allowed to have ANY download permissions.  Again, maybe not possible with respect to the orig SLSK client aspects, but with respect to N+.

--- DS

**Reply by mathiascode** [reply]
> how about if someone lists something as PRIVATE, it ISN'T SHARED by default?

Unless I'm misunderstanding you, that's how it works by default. People can optionally enable the setting to show privately shared files to everyone (except for users who have hidden them in their client).

> Also, if someone has their ENTIRE share (or even over 50% of their share!) as PRIVATE, they should NOT be allowed to have ANY download permissions.

I don't think arbitrary, enforced limits are the solution. If you start requiring a certain number of public files, you shut out anyone new to Soulseek who doesn't have an existing collection of files.

It's similar to an "auto-ban" feature, which I don't want anything to do with due to all the drama surrounding it.

**Reply by mathiascode** [reply]
Arbitrary limits would also likely result in users sharing a certain number of garbage files in order to get past them, polluting the network.

**Reply by DeathStalker77** [reply]
No, I'm not suggesting any kind of minimum.  I'm just talking about **trying** to shut out as many of these a-holes as possible.  I want the burden put on THEM, not good users - WE shouldn't have to select "Hide Private Files" - THEY should not be allowed to have them show up to begin with.  It's a taunt, a "power trip" for most of them - like a kindergarten-"I have this and you can't have it"-game to them.  The less they are able to do that, the more likely they are to actually share.  Again, I'm not talking about having minimum shares, I'm talking about not being able to have your entire "share" as PRIVATE - that is simply NOT "sharing".  JMHO.

**Reply by slook** [reply]
Maybe it could be a viable idea to have some kind of alert overlay/popup message if the shares are configured in such a way that makes all of them private, such as, if all of the shared folders are set to Buddies and there are no Public folders (or there are no shares defined whatsoever).

Perhaps a confirmation message box popup to confirm closing the Preferences dialog (or moving on to the next page in Setup Assistant), could prompt the user about such an incomplete/incorrect configuration potentially requiring attention and bring them to the relevant settings page.
